msdbook/tests/test_utils.py

```diff
@@ -0,0 +1,119 @@
+import pytest
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from msdbook.utils import fit_logit, plot_contour_map
+from statsmodels.base.wrapper import ResultsWrapper
+
+@pytest.fixture
+def sample_data():
+    """Fixture to provide sample data for testing."""
+    np.random.seed(0)
+    # Generate some random data
+    n = 100
+    df = pd.DataFrame({
+        'Success': np.random.randint(0, 2, size=n),
+        'Predictor1': np.random.randn(n),
+        'Predictor2': np.random.randn(n),
+        'Interaction': np.random.randn(n)
+    })
+    return df
+
+def test_fit_logit(sample_data):
+    """Test the fit_logit function."""
+    predictors = ['Predictor1', 'Predictor2']
+    result = fit_logit(sample_data, predictors)
+    
+    # Check if result is a statsmodels LogitResultsWrapper object
+    assert isinstance(result, ResultsWrapper) 
+    
+    # Check if the result object has the expected attributes
+    assert hasattr(result, 'params')
+    assert hasattr(result, 'pvalues')
+    assert hasattr(result, 'predict')
+
+    # Check that parameters (coefficients) are not empty
+    assert result.params is not None
+    assert result.pvalues is not None
+
+def test_plot_contour_map(sample_data):
+    """Test the plot_contour_map function."""
+    fig, ax = plt.subplots()
+    
+    # Fit a logit model for the purpose of plotting
+    predictors = ['Predictor1', 'Predictor2']
+    result = fit_logit(sample_data, predictors)
+    
+    xgrid = np.linspace(-2, 2, 50)
+    ygrid = np.linspace(-2, 2, 50)
+    
+    contour_cmap = 'viridis'
+    dot_cmap = 'coolwarm'
+    levels = np.linspace(0, 1, 10)
+    
+    # Call the plot function
+    contourset = plot_contour_map(
+        ax, result, sample_data, contour_cmap, dot_cmap, levels, xgrid, ygrid, 'Predictor1', 'Predictor2', base=0
```

Comment:
you should avoid hardcoding values as much as possible

Reply:
I should have been clearer here, for tests you want to have hardcoded values as inputs and outputs so that you can make sure the function is doing the right thing, but in general, if you're using the same value over and over, you should put it into a variable and use that variable. For example, you use `'Predictor1'` and `'Predictor2'` a lot so you should put those into variables.